pandas/core/generic.py

```diff
@@ -2319,7 +2319,11 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
             other views on this object, (e.g. a no-copy slice for a column in a
             DataFrame).
         limit : int, default None
-            Maximum size gap to forward or backward fill
+            If method is specified, this is the maximum number of consecutive
+            NaN values to forward/backward fill. In other words, if there is
+            a gap with more than this number of consecutive NaNs, it will only
+            be partially filled. If method is not specified, this is the
+            maximum number of entries along the axis where NaNs will be filled.
```

Comment:
one small nitpick: maybe "along the *entire* axis"

For the rest everything clear now! Thanks!